gammapy/scripts/image_pipe.py

```diff
@@ -2,56 +2,180 @@
+    counts_image : ~gammapy.image.SkyMap`
+            count image
+    bkg_image : ~gammapy.image.SkyMap`
+            bkg image
     """
 
-    def __init__(self, config):
-        self.config = config
-
-    @classmethod
-    def from_yaml(cls, filename):
-        """Read config from YAML file."""
-        import yaml
-        log.info('Reading {}'.format(filename))
-        with open(filename) as fh:
-            config = yaml.safe_load(fh)
-        return cls(config)
-
-    def run(self):
-        """Run analysis chain."""
-        log.info('Running analysis ...')
-        print(self.config['general']['outdir'])
-        print(self.config)
-
-    def make_counts(self):
+    def __init__(self, empty_image=None,
+                 energy_band=None, offset_band=None,
+                 data_store=None, counts_image=None, bkg_image=None):
+        self.maps = SkyMapCollection()
+        if counts_image:
+            self.maps["total_counts"] = counts_image
+        else:
+            self.data_store = data_store
+            self.energy_band = energy_band
+            self.offset_band = offset_band
+
+            self.empty_image = empty_image
+            self.header = self.empty_image.to_image_hdu().header
+        if bkg_image:
+            self.maps["total_bkg"] = bkg_image
+
+    def make_counts(self, obs_id):
+        """Fill the counts image for the events of one observation.
+
+        Parameters
+        ----------
+        obs_id : int
+            Number of the observation
+
+        """
+        self.maps["counts"] = SkyMap.empty_like(self.empty_image)
+        obs = self.data_store.obs(obs_id=obs_id)
+        events = obs.events.select_energy(self.energy_band)
+        events = events.select_offset(self.offset_band)
+        self.maps["counts"].fill(events=events)
+        self.maps["counts"].data = self.maps["counts"].data.value
         log.info('Making counts image ...')
 
-    def make_exposure(self):
-        log.info('Making exposure image ...')
+    def make_total_counts(self):
+        """Stack the total count from the observation in the 'DataStore'
+
+        """
+        self.maps["total_counts"] = SkyMap.empty_like(self.empty_image)
+        for obs_id in self.data_store.obs_table['OBS_ID']:
+            self.make_counts(obs_id)
+            self.maps["total_counts"].data += self.maps["counts"].data
+
+    def make_background(self, obs_id):
+        """Make the background map for one observation from a bkg model
+
+        Parameters
+        ----------
+        obs_id : int
+            Number of the observation
+
+        """
+        self.maps["bkg"] = SkyMap.empty_like(self.empty_image)
+        obs = self.data_store.obs(obs_id=obs_id)
+        table = obs.bkg.acceptance_curve_in_energy_band(energy_band=self.energy_band)
+        center = obs.pointing_radec.galactic
+        bkg_hdu = fill_acceptance_image(self.header, center, table["offset"], table["Acceptance"], self.offset_band[1])
```

Comment:
It might be worth, changing the `fill_acceptance_image` to work with a `SkyMap` object instead of an `ImageHDU`.
This will get rid of a some code and further unify the gammapy API.